gnovm/tests/files/alloc_8.gno

```diff
@@ -0,0 +1,12 @@
+// MAXALLOC: 50000000
+package main
+
+func recurse(n int) {
+	recurse(n + 1)
+}
+func main() {
+	recurse(0)
+}
+
+// Error:
+// should not happen, allocation limit exceeded while gc.
```

Comment:
I don't get it, why does this happen in GC?

Reply:
in short, the "should not happen" is should not happen, it is a corner case where the GC count exceeds the number present when GC starts, which is incorrect. The new commit: https://github.com/gnolang/gno/pull/4302#issuecomment-3244393915 fix this by e.g. adding missed allocations and refining GC counting strategies, etc.